pkg/api/v1/zz_generated.openapi.validation.go

```diff
@@ -0,0 +1,214 @@
+*/
+
+// This file was autogenerated by openapi-gen. Do not edit it manually!
+
+package v1
+
+import (
+	field "k8s.io/kubernetes/pkg/util/validation/field"
+	validation "k8s.io/kubernetes/pkg/validation"
+)
+
+func (s Binding) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s ComponentStatus) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s ConfigMap) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s Endpoints) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s Event) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s LimitRange) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s Namespace) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s Node) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if errs := s.ObjectMeta.Validate(&validation.FieldMeta{Path: meta.Child("ObjectMeta")}, op); len(errs) != 0 {
+		allErrs = append(allErrs, errs...)
+	}
+	return allErrs
+}
+
+func (s ObjectMeta) Validate(meta *validation.FieldMeta, op validation.OperationType) field.ErrorList {
+	allErrs := field.ErrorList{}
+	{ // Validate: Len(5,100)
+		v := &validation.Len{}
+		errs := v.Init(op, &validation.FieldMeta{Path: meta.Child("Name"), Type: "string"}, 5, 100)
```

Comment:
I don't understand the "Type" field here.  This feels a little complex for something as simple as length-check - why is it not a call to Len(s.Name, min, max, fldPath)?

I know that OneOf is WAY more complicated than Len, but my mind goes to all the other crazy corner cases we will hit.  How do I add new one-use validations without having to extend the generator?

Can we find a balance where simple cases are simply function calls and complex cases require more complex setup?